image-loader/app/lib/ImageProcessorClassReflector.scala

```diff
@@ -0,0 +1,47 @@
+package lib
+
+import com.gu.mediaservice.lib.cleanup.{ImageProcessor, ImageProcessorConfig}
+import com.gu.mediaservice.lib.config.CommonConfig
+import com.typesafe.scalalogging.StrictLogging
+
+import scala.reflect.runtime.universe
+import scala.util.control.NonFatal
+
+object ImageProcessorClassReflector extends StrictLogging {
+  def catchNonFatal[T](block: => T): Either[Throwable, T] = {
+    try {
+      Right(block)
+    } catch {
+      case NonFatal(e) => Left(e)
+    }
+  }
+
+  def loadImageProcessor(className: String, config: ImageProcessorConfig): Either[String, ImageProcessor] = {
+    val mirror = universe.runtimeMirror(getClass.getClassLoader)
+
+    try {
+      for {
+        classSymbol <- catchNonFatal(mirror.staticClass(className)).left.map(_ => s"Class $className not found")
+        isImageProcessor = classSymbol.baseClasses.exists(symbol => symbol.fullName == classOf[ImageProcessor].getCanonicalName)
+        _ <- if (isImageProcessor) Right(()) else Left(s"Configured image processor $className does not extend ${classOf[ImageProcessor].getCanonicalName}")
+        classMirror = mirror.reflectClass(classSymbol)
+        constructor = classSymbol.primaryConstructor.asMethod
+        params = constructor.paramLists.head
+        hasConfigParam = {
+          val configParamType = Class.forName(params.head.info.toString, true, getClass.getClassLoader)
+          val assignable = classOf[ImageProcessorConfig].isAssignableFrom(configParamType)
+          System.err.println(s"${params.length} ${classOf[CommonConfig].getCanonicalName} ${configParamType} ${assignable}")
```

Comment:
Is logging not set up yet?

Reply:
Hah - that was accidentally left in from testing in the console where logging wasn't initialised. Will remove.